js/src/logger.ts

```diff
@@ -406,15 +413,153 @@ interface UserInfo {
+   * @param event.scores: A dictionary of numeric values (between 0 and 1) to log. The scores should give you a variety of signals that help you determine how accurate the outputs are compared to what you expect and diagnose failures. For example, a summarization app might have one score that tells you how accurate the summary is, and another that measures the word similarity between the generated and grouth truth summary. The word similarity score could help you determine whether the summarization was covering similar concepts or not. You can use these scores to help you sort, filter, and compare logs.
+   * @param event.metadata: (Optional) a dictionary with additional data about the test example, model outputs, or just about anything else that's relevant, that you can use to help find and analyze examples later. For example, you could log the `prompt`, example's `id`, or anything else that would be useful to slice/dice later. The values in `metadata` can be any JSON-serializable type, but its keys must be strings.
+   * @param event.metrics: (Optional) a dictionary of metrics to log. The following keys are populated automatically and should not be specified: "start", "end".
+   * @param event.id: (Optional) a unique identifier for the event. If you don't provide one, BrainTrust will generate one for you.
+   * :returns: The `id` of the logged event.
+   */
+  public async log(event: Readonly<ExperimentLogPartialArgs>): Promise<string> {
+    const span = await this.startSpan({ startTime: this.lastStartTime, event });
+    this.lastStartTime = span.end();
+
+    if (!this.logOptions.asyncFlush) {
+      await this.flush();
+    }
+
+    return span.id;
+  }
+
+  /**
+   * Create a new toplevel span. The name parameter is optional and defaults to "root".
+   *
+   * See `Span.startSpan` for full details.
+   */
+  public async startSpan(args?: StartSpanOptionalNameArgs): Promise<Span> {
+    await this.lazyLogin();
+    const project = await this.lazyProject.lazyInit();
+    const { name, ...argsRest } = args ?? {};
+    return new SpanImpl({
+      bgLogger: this.bgLogger,
+      name: name ?? "root",
+      ...argsRest,
+      rootProject: project,
+    });
+  }
+
+  /**
+   * Wrapper over `Logger.startSpan`, which passes the initialized `Span` it to the given callback and ends it afterwards. See `Span.traced` for full details.
+   */
+  public async traced<R>(
+    callback: (span: Span) => R,
+    args?: StartSpanOptionalNameArgs & SetCurrentArg
+  ): Promise<R> {
+    const { setCurrent, ...argsRest } = args ?? {};
+    const span = await this.startSpan(argsRest);
+    try {
+      let ret = null;
```

Comment:
Why can't this just be
```
return await ((setCurrent ?? true) ? withCurrent(span, () => callback(span)) : callback(span));
```

Reply:
I guess it can, since awaiting non-promises just returns the value